xDrip/Scenes/Settings/AlertSettings/SettingsAlertSingleType/SettingsAlertSingleTypePresenter.swift

```diff
@@ -22,7 +23,201 @@ final class SettingsAlertSingleTypePresenter: SettingsAlertSingleTypePresentatio
+            guard let string = string else { return }
+            editingChangedHandler(string)
+        }
+    }
+    
+    private func createRightSwitchCell(
+        _ field: SettingsAlertSingleType.Field,
+        isSwitchOn: Bool,
+        switchValueChangedHandler: @escaping (SettingsAlertSingleType.Field, Bool) -> Void) -> BaseSettings.Cell {
+        
+        return .rightSwitch(text: field.title, isSwitchOn: isSwitchOn) { (value) in
+            switchValueChangedHandler(field, value)
+        }
+    }
+    
+    private func createCountDownPickerView(
+        _ field: SettingsAlertSingleType.Field,
+        detailText: String?,
+        valueChangeHandler: @escaping (SettingsAlertSingleType.Field, TimeInterval) -> Void) -> BaseSettings.Cell {
+        
+        let picker = CustomDatePicker()
+        
+        DispatchQueue.main.async {
+            picker.datePickerMode = .countDownTimer
+            picker.countDownDuration = 0
+            picker.minuteInterval = 1
+        }
+        
+        picker.formatDate = { date in
+            let components = Calendar.current.dateComponents([.hour, .minute], from: date)
+            
+            guard let hour = components.hour, let minute = components.minute else { return " " }
+            
+            let time = minute + hour * 60
+            
+            valueChangeHandler(field, TimeInterval(time))
+            
+            return "\(time) m"
+        }
+        
+        return .pickerExpandable(mainText: field.title, detailText: detailText, picker: picker)
+    }
+    
+    private func createTimePickerView(
+        _ field: SettingsAlertSingleType.Field,
+        date: Date?,
+        detailText: String?,
+        valueChangeHandler: @escaping (SettingsAlertSingleType.Field, TimeInterval) -> Void) -> BaseSettings.Cell {
+        
+        let picker = CustomDatePicker()
+        
+        picker.datePickerMode = .time
+        
+        if let date = date {
+            picker.date = date
+        }
+        
+        picker.formatDate = { date in
+            let components = Calendar.current.dateComponents([.hour, .minute], from: date)
+            guard let date = Calendar.current.date(from: components) else { return " " }
+            
+            valueChangeHandler(field, TimeInterval(date.timeIntervalSince1970))
+            
+            return DateFormatter.localizedString(from: date, dateStyle: .none, timeStyle: .short)
+        }
+        
+        return .pickerExpandable(mainText: field.title, detailText: detailText, picker: picker)
+    }
+    
+    private func createUnitsPickerView(
+        _ field: SettingsAlertSingleType.Field,
+        detailText: String?,
+        valueChangeHandler: @escaping (SettingsAlertSingleType.Field, Float) -> Void) -> BaseSettings.Cell {
+        
+        let range = User.current.settings.unit.rangeValues
+        let step: Float = User.current.settings.unit == .mmolL ? 0.1 : 1.0
```

Comment:
I think such stepping will be used in other places as well. Let's move it to `GlucoseUnit`.